ports/krb5/portfile.cmake

```
@@ -0,0 +1,32 @@
+vcpkg_download_distfile(ARCHIVE
+    URLS "https://kerberos.org/dist/krb5/1.21/krb5-${VERSION}.tar.gz"
+    FILENAME "krb5-${VERSION}.tar.gz"
+    SHA512 4e09296b412383d53872661718dbfaa90201e0d85f69db48e57a8d4bd73c95a90c7ec7b6f0f325f6bc967f8d203b256b071c0191facf080aca0e2caec5d0ac49
+)
+
+vcpkg_extract_source_archive(
+    SOURCE_PATH
+    ARCHIVE ${ARCHIVE}
+)
+
+vcpkg_configure_make(
+    SOURCE_PATH "${SOURCE_PATH}/src"
+    AUTOCONFIG
+    OPTIONS
+    CFLAGS=-fcommon
```

Doesn't this override `CFLAGS` completely?

---

I think not, How can verify it?

---

from build log:
```
/usr/bin/cc -DHAVE_CONFIG_H  -I../../include -I../.././../src/krb5-1-8a38cd677f.clean/src/include  -DKRB5_DEPRECATED=1 -DKRB5_PRIVATE -DNDEBUG -fcommon -Wall -Wcast-align -Wshadow -Wmissing-prototypes -Wno-format-zero-length -Woverflow -Wstrict-overflow -Wmissing-format-attribute -Wmissing-prototypes -Wreturn-type -Wmissing-braces -Wparentheses -Wswitch -Wunused-function -Wunused-label -Wunused-variable -Wunused-value -Wunknown-pragmas -Wsign-compare -Werror=uninitialized -Wno-maybe-uninitialized -Werror=pointer-arith -Werror=int-conversion -Werror=incompatible-pointer-types -Werror=discarded-qualifiers -Werror=implicit-int -Werror=declaration-after-statement -Werror-implicit-function-declaration -pthread  -c ../.././../src/krb5-1-8a38cd677f.clean/src/util/support/threads.c
```

---

So this really overrides essential flags from vcpkg, in particular `-fPIC`, `-g`/`-O3`.
Try 
```suggestion
        "CFLAGS=-fcommon \$CFLAGS"
```

But why is the extra flag needed at all?

---

without the flag:
```
/usr/bin/ld: ../../lib/libkadm5srv_mit.a(server_kdb.o):/home/tal/vcpkg/buildtrees/krb5/x64-linux-dbg/lib/kadm5/srv/../../.././../src/krb5-1-8a38cd677f.clean/src/lib/kadm5/srv/server_kdb.c:18: multiple definition of `master_keyblock'; kdb5_util.o:/home/tal/vcpkg/buildtrees/krb5/x64-linux-dbg/kadmin/dbutil/../.././../src/krb5-1-8a38cd677f.clean/src/kadmin/dbutil/kdb5_util.c:109: first defined here
collect2: error: ld returned 1 exit status
```

---

I see, really old (> 10 years) C code, and gcc 10 making '-fno-common` the default.

---

> So this really overrides essential flags from vcpkg, in particular `-fPIC`, `-g`/`-O3`. Try

Thank you for your suggestion, it help!
```
/usr/bin/cc -DHAVE_CONFIG_H  -I../../include -I../.././../src/krb5-1-8a38cd677f.clean/src/include -I../.././../src/krb5-1-8a38cd677f.clean/src/lib/gssapi/generic -I../.././../src/krb5-1-8a38cd677f.clean/src/lib/gssapi/krb5 -I../../lib/gssapi/generic -I../../lib/gssapi/krb5 -DKRB5_DEPRECATED=1 -DKRB5_PRIVATE  -fcommon -fPIC -g -Wall -Wcast-align -Wshadow -Wmissing-prototypes -Wno-format-zero-length -Woverflow -Wstrict-overflow -Wmissing-format-attribute -Wmissing-prototypes -Wreturn-type -Wmissing-braces -Wparentheses -Wswitch -Wunused-function -Wunused-label -Wunused-variable -Wunused-value -Wunknown-pragmas -Wsign-compare -Werror=uninitialized -Wno-maybe-uninitialized -Werror=pointer-arith -Werror=int-conversion -Werror=incompatible-pointer-types -Werror=discarded-qualifiers -Werror=implicit-int -Werror=declaration-after-statement -Werror-implicit-function-declaration -pthread  -c ../.././../src/krb5-1-8a38cd677f.clean/src/kadmin/server/auth.c
```